src/main/java/com/direwolf20/buildinggadgets/common/building/placement/ExclusiveAxisChasing.java

```diff
@@ -0,0 +1,84 @@
+package com.direwolf20.buildinggadgets.common.building.placement;
+
+import com.direwolf20.buildinggadgets.common.building.IPlacementSequence;
+import com.direwolf20.buildinggadgets.common.building.Region;
+import com.direwolf20.buildinggadgets.common.tools.VectorTools;
+import com.google.common.collect.AbstractIterator;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumFacing.Axis;
+import net.minecraft.util.EnumFacing.AxisDirection;
+import net.minecraft.util.math.BlockPos;
+
+import javax.annotation.Nonnull;
+import java.util.Iterator;
+
+/**
+ * Starts from the selected position, and extend a column of blocks towards player on the axis of the selected face.
+ */
+public final class ExclusiveAxisChasing implements IPlacementSequence {
+
+    public static ExclusiveAxisChasing create(BlockPos source, BlockPos target, Axis axis) {
+        int difference = VectorTools.getAxisValue(target, axis) - VectorTools.getAxisValue(source, axis);
+        if (difference < 0)
+            return create(source, target, EnumFacing.getFacingFromAxis(AxisDirection.NEGATIVE, axis));
+        return create(source, target, EnumFacing.getFacingFromAxis(AxisDirection.POSITIVE, axis));
+    }
+
+    /**
+     * <p>Note that this factory method does not ensure {@code offsetDirection} is appropriate. Use {@link #create(BlockPos, BlockPos, Axis)} if this is required.</p>
+     */
+    public static ExclusiveAxisChasing create(BlockPos source, BlockPos target, EnumFacing offsetDirection) {
+        Axis axis = offsetDirection.getAxis();
+        int difference = VectorTools.getAxisValue(target, axis) - VectorTools.getAxisValue(source, axis);
+        // int maxProgression = MathHelper.clamp(Math.abs(difference), 1, Integer.MAX_VALUE);
```

Comment:
If it's not being used then remove it or add a comment explaining why it's not currently being used